oop2-encapsulate-inherit.js

```diff
@@ -1,45 +1,346 @@
+    this.#books.push([...newBooks]);
+  }
+  addReview(review){
+    this.#reviews.push(review)
+  }
+}
 
+class Publisher {
+  #authors = [];
+  #books = [];
+  constructor(authors, books) {
+    // (this.#authors = authors), (this.#books = books);
+  }
+  //getters
+  getAuthors() {
+    return this.#authors;
+  }
+  getBooks() {
+    return this.#books;
+  }
+  //setters
+  setAuthor(newAuths) {
+    this.#authors.push(newAuths);
+  }
+  setBooks(newBooks) {
+    this.#books.push([...newBooks]);
+  }
+}
 
+class Review {
+  #rating;
+  #user;
+  constructor(rating, user) {
+    (this.#rating = rating), (this.#user = user);
+  }
+  //getters
+  getRating() {
+    return this.#rating;
+  }
+  getUser() {
+    return this.#user;
+  }
+  //setters
+  setRating(newRating) {
+    this.#rating = newRating;
+  }
+  setUser(newUser) {
+    this.#user = newUser;
+  }
+}
 
+// 2. Create the following classes:
+//     - Umbrella class.  It should have a an organization name.  It should only have a single instance.  Umbrella is the "Parent Organization" e.g. Allegis Group
+//     - Company class. It should have a compnay name and # of employees.  Create at least 3 different companies that are children of the Umbrella Organization.  (e.g. TEKsystems, Aerotek, etc.)
+//     - Site class. It should have the name of the company, and the location of the site.  It is a child class that inherits from the Company class.  Create 3 sites for each company.
+//     - Employee class. It should have a employee name, job title, and salary properties. Create 10 employee instances: 1 CEO, 1 Manager, 1 Secretary, 2 Engineers, 1 Financial Officer, 1 Janitor, 1 Marketer, 1 HR Personnel, 1 Lawyer.  Feel free to add other employee roles at your own discretion.
 
-2. Create the following classes:
-    - Umbrella class.  It should have a an organization name.  It should only have a single instance.  Umbrella is the "Parent Organization" e.g. Allegis Group
-    - Company class. It should have a compnay name and # of employees.  Create at least 3 different companies that are children of the Umbrella Organization.  (e.g. TEKsystems, Aerotek, etc.)
-    - Site class. It should have the name of the company, and the location of the site.  It is a child class that inherits from the Company class.  Create 3 sites for each company.
-    - Employee class. It should have a employee name, job title, and salary properties. Create 10 employee instances: 1 CEO, 1 Manager, 1 Secretary, 2 Engineers, 1 Financial Officer, 1 Janitor, 1 Marketer, 1 HR Personnel, 1 Lawyer.  Feel free to add other employee roles at your own discretion.
-    
-    - For each class, add 2 properties and 2 methods to each.  The properties and methods should make sense, considering what the classes are supposed to be representing.
+//     - For each class, add 2 properties and 2 methods to each.  The properties and methods should make sense, considering what the classes are supposed to be representing.
 
+class Umbrella {
```

Comment:
I like how you approached this, by using properties from previously-created objects (FYI, this approach was focused on a learning objective so we're not being overly-proper, but in the real world, it would have been a has-a relationship, not is-a, because now every employee is an Umbrella)